code/Control/UserDefinedFormController.php

```diff
@@ -104,10 +104,13 @@ protected function addUserFormsValidatei18n()
      * where the form should be rendered into. If it does not exist
      * then default back to $Form.
      *
-     * @return array
+     * @return array|Object
```

Comment:
Hmm, maybe we should maintain return type... Though I'm not sure. But this definitely should not be Object, Controller would be better

Reply:
I'm unsure also.

1. method never used to return anything other than array.
2. this method is an implementation of the RequestHandler interface (extremely loose as it's defined).

Which leads us to

1. changing this makes it a backwards compatibility thing, major version movement.
2. Returning anything expected by `RequestHandler::handleAction` is fine.